pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -231,6 +231,14 @@ func SetDefaults_Shoot(obj *Shoot) {
 		if obj.Spec.Kubernetes.Kubelet.FailSwapOn == nil {
 			obj.Spec.Kubernetes.Kubelet.FailSwapOn = pointer.Bool(true)
 		}
+		nodeSwapFeatureGateEnabled, ok := obj.Spec.Kubernetes.Kubelet.FeatureGates["NodeSwap"]
+		if ok && nodeSwapFeatureGateEnabled && !*obj.Spec.Kubernetes.Kubelet.FailSwapOn && obj.Spec.Kubernetes.Kubelet.MemorySwap == nil {
+			obj.Spec.Kubernetes.Kubelet.MemorySwap = &MemorySwapConfiguration{}
+		}
+		if ok && nodeSwapFeatureGateEnabled && !*obj.Spec.Kubernetes.Kubelet.FailSwapOn && obj.Spec.Kubernetes.Kubelet.MemorySwap.SwapBehavior == nil {
+			limitedSwap := LimitedSwap
+			obj.Spec.Kubernetes.Kubelet.MemorySwap.SwapBehavior = &limitedSwap
+		}
```

Comment:
```suggestion
		if nodeSwapFeatureGateEnabled, ok := obj.Spec.Kubernetes.Kubelet.FeatureGates["NodeSwap"]; ok && nodeSwapFeatureGateEnabled {
			if !*obj.Spec.Kubernetes.Kubelet.FailSwapOn && obj.Spec.Kubernetes.Kubelet.MemorySwap == nil {
				obj.Spec.Kubernetes.Kubelet.MemorySwap = &MemorySwapConfiguration{}
			}
			if  !*obj.Spec.Kubernetes.Kubelet.FailSwapOn && obj.Spec.Kubernetes.Kubelet.MemorySwap.SwapBehavior == nil {
				limitedSwap := LimitedSwap
				obj.Spec.Kubernetes.Kubelet.MemorySwap.SwapBehavior = &limitedSwap
			}
		}
```

same for `v1beta1`

Reply:
What if the feature gate is only enabled for a specific worker pool?

same for `v1beta1`

Reply:
Also, the defaulting is only done for k8s >= 1.22, because that's when the swap behavior got configurable.